src/accountsdb/fuzz.zig

```diff
@@ -45,56 +49,93 @@ pub fn run(seed: u64, args: *std.process.ArgIterator) !void {
         }
     };
 
-    var gpa_allocator = std.heap.GeneralPurposeAllocator(.{}){};
-    const allocator = gpa_allocator.allocator();
+    var prng = std.Random.DefaultPrng.init(seed);
+    const rand = prng.random();
+
+    var gpa_state = std.heap.GeneralPurposeAllocator(.{}){};
+    defer _ = gpa_state.deinit();
+    const gpa = gpa_state.allocator();
 
-    const logger = Logger.init(allocator, .debug);
+    const logger = Logger.init(gpa, .debug);
     defer logger.deinit();
     logger.spawn();
 
-    var prng = std.rand.DefaultPrng.init(seed);
-    const rand = prng.random();
-
     const use_disk = rand.boolean();
-    const snapshot_dir = "test_data/accountsdb_fuzz";
+
+    var test_data_dir = try std.fs.cwd().makeOpenPath("test_data", .{});
+    defer test_data_dir.close();
+
+    const snapshot_dir_name = "accountsdb_fuzz";
+    var snapshot_dir = try test_data_dir.makeOpenPath(snapshot_dir_name, .{});
+    defer snapshot_dir.close();
     defer {
         // NOTE: sometimes this can take a long time so we print when we start and finish
         std.debug.print("deleting snapshot dir...\n", .{});
-        std.fs.cwd().deleteTree(snapshot_dir) catch |err| {
-            std.debug.print("failed to delete snapshot dir: {}\n", .{err});
+        test_data_dir.deleteTree(snapshot_dir_name) catch |err| {
+            std.debug.print("failed to delete snapshot dir ('{s}'): {}\n", .{ sig.utils.fmt.tryRealPath(snapshot_dir, "."), err });
         };
         std.debug.print("deleted snapshot dir\n", .{});
     }
     std.debug.print("use disk: {}\n", .{use_disk});
 
-    var accounts_db = try AccountsDB.init(allocator, logger, .{
+    const alt_dir_name = "alt";
+    var alt_dir = try snapshot_dir.makeOpenPath(alt_dir_name, .{});
```

Comment:
`alt_dir` -> `alternative_snapshot_dir` - can we also add a comment explaining some context - ie, we need to pack + unpack a snapshot for full testing and we dont want to do this in the same accounts/ directory

Reply:
12017298089cacbd8be261f45b19c44b2fd27e42